### PR TITLE
omnibus: store config.guess output files in the regular install directory

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR reverts an old change that was storing config.guess files in `/tmp/build`

### Motivation

While we don't want to ship config.guess files as they're only relevant during the build, storing them in /tmp prevents omnibus cache to handle them, causing build failures as soon as they are required.

### Additional Notes

This PR actually doesn't change anything directly, but relies on 2 other PRs:
* https://github.com/DataDog/omnibus-software/pull/473 to revert back to the old installation dir
* https://github.com/DataDog/omnibus-ruby/pull/157 to update the expected path for `update_config_guess` directive
 and simply removes config.guess from the install directory before packaging the agent.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by pgimalac — APPROVED** [review]

File: release.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
     },
     "nightly": {
         "INTEGRATIONS_CORE_VERSION": "master",
-        "OMNIBUS_SOFTWARE_VERSION": "master",
-        "OMNIBUS_RUBY_VERSION": "datadog-5.5.0",
+        "OMNIBUS_SOFTWARE_VERSION": "chouquette/config_guess",
+        "OMNIBUS_RUBY_VERSION": "chouquette/config_guess",
```
Comment (Pythyu) [inline release.json:10]: nitpick to not forget to put `master` and `datadog-5.5.0`

File: release.json [hunk 2]
```diff
@@ -18,8 +18,8 @@
     },
     "nightly-a7": {
         "INTEGRATIONS_CORE_VERSION": "master",
-        "OMNIBUS_SOFTWARE_VERSION": "master",
-        "OMNIBUS_RUBY_VERSION": "datadog-5.5.0",
+        "OMNIBUS_SOFTWARE_VERSION": "chouquette/config_guess",
+        "OMNIBUS_RUBY_VERSION": "chouquette/config_guess",
```
Comment (Pythyu) [inline release.json:22]: Same as above

**Review by Pythyu — APPROVED** [review]
A small nitpick but LGTM

File: release.json (hunk 2 above)
Comment (chouquette) [reply release.json:22]: Ah, that's why I initially opened it as a draft, and I forgot about it after Pierre mentionned it :facepalm:

Comment (chouquette) [reply release.json]: Thanks, without your nit I'd probably have commited the change as is :sweat_smile:

**Pull request merged**